modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/idp/mgt/IdentityProviderMgtServiceTestCase.java

```diff
@@ -360,7 +365,202 @@ public void testGetAllIdPs() throws Exception {
+                    "empty filter");
+            Assert.assertEquals(filteredIdpSize, 2, "Unable to find filtered identity provider while testing " +
+                    "getPaginatedIdPsInfo() with empty filter");
+        } else {
+            Assert.fail("Unable to find filtered identity provider while testing getPaginatedIdPsInfo().");
+        }
+    }
+
+    @Test(priority = 9, groups = "wso2.is", description = "test getPaginatedIdPsInfo operation with invalid filter")
+    public void testGetPaginatedIdPsInfoWithInvalidFilter() {
+
+        try {
+            String filter = "names sp \"Search\" and namees ew \"Test\"";
+            int pageNumber = 1;
+            List<IdentityProvider> idpList = idpMgtServiceClient.getPaginatedIdPsInfo(filter, pageNumber);
+            Assert.assertNull(idpList, "Idp list found while calling getPaginatedIdPsInfo() with invalid filter.");
+        } catch (Exception e) {
+            Assert.assertEquals(e.getMessage(), invalidFilterErrorMessage,
+                    "Error occurred while testing testGetPaginatedIdPsInfoWithInvalidFilter() with invalid filter.");
+        }
+    }
+
+    @Test(priority = 10, groups = "wso2.is", description = "test getPaginatedIdPsInfo operation with invalid page number")
+    public void testGetPaginatedIdPsInfoWithInvalidPageNumber() {
+
+        try {
+            String filter = "name sw \"Search\" and name ew \"Test\"";
+            int pageNumber = -1;
+            List<IdentityProvider> idpList = idpMgtServiceClient.getPaginatedIdPsInfo(filter, pageNumber);
+            Assert.assertNull(idpList,
+                    "List of Idps found  while calling getPaginatedIdPsInfo() with invalid page number");
+        } catch (Exception e) {
+            Assert.assertEquals(e.getMessage(), invalidPageNumberErrorMessage, "Error occurred while testing " +
+                    "testGetPaginatedIdPsInfoWithInvalidPageNumber() with invalid page number.");
+        }
+    }
+
+    @Test(priority = 11, groups = "wso2.is", description = "test getAllPaginatedIdPsInfo operation with incorrect " +
+            "page number")
+    public void testGetPaginatedIdPsInfoWithIncorrectPageNumber() throws Exception {
+
+        String filter = "name sw \"Search\" and name ew \"Test\"";
+        int pageNumber = 2;
+        List<IdentityProvider> idpList = idpMgtServiceClient.getPaginatedIdPsInfo(filter, pageNumber);
+        if (idpList.size() > 0) {
+            Assert.fail("Invalid idp list found while calling getPaginatedIdPsInfo() with incorrect page number.");
+        }
+    }
+
+    @Test(priority = 12, groups = "wso2.is", description = "test getAllIdpCount operation")
+    public void testGetAllIdPsCount() throws Exception {
+
+        int idpCount = idpMgtServiceClient.getAllIdpCount();
+        log.info("All idp paginated filtered count : " + idpCount);
+        Assert.assertEquals(idpCount, 2, "Total Idp Count not found ");
```

Comment:
change assert failure message as, Total idp count did not match with the expected